android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginViewModel.kt

```diff
@@ -1,33 +1,47 @@
 package vn.luongvo.kmm.survey.android.ui.screens.login
 
 import androidx.lifecycle.viewModelScope
-import kotlinx.coroutines.flow.catch
-import kotlinx.coroutines.launch
-import timber.log.Timber
+import kotlinx.coroutines.flow.*
 import vn.luongvo.kmm.survey.android.ui.base.BaseViewModel
-import vn.luongvo.kmm.survey.data.local.datasource.TokenLocalDataSource
+import vn.luongvo.kmm.survey.android.ui.navigation.AppDestination
+import vn.luongvo.kmm.survey.domain.usecase.IsLoggedInUseCase
 import vn.luongvo.kmm.survey.domain.usecase.LogInUseCase
 
 class LoginViewModel(
-    private val logInUseCase: LogInUseCase,
-    private val tokenLocalDataSource: TokenLocalDataSource
+    private val isLoggedInUseCase: IsLoggedInUseCase,
+    private val logInUseCase: LogInUseCase
 ) : BaseViewModel() {
 
-    fun logIn() {
-        // TODO sample request to verify network layer implementation
-        viewModelScope.launch {
-            logInUseCase(
-                email = "luong@nimblehq.co",
-                password = "12345678"
-            )
-                .catch { e ->
-                    Timber.e(e)
-                }
-                .collect {
-                    Timber.d("tokenType=${tokenLocalDataSource.tokenType}")
-                    Timber.d("accessToken=${tokenLocalDataSource.accessToken}")
-                    Timber.d("refreshToken=${tokenLocalDataSource.refreshToken}")
+    private val _isLoggedIn = MutableStateFlow<Boolean?>(null)
+    val isLoggedIn: StateFlow<Boolean?> = _isLoggedIn
+
+    fun init() {
+        isLoggedInUseCase()
+            .catch { e -> _error.emit(e) }
```

Comment:
@luongvo I think catching the error on this case is redundant 🤔

Reply:
@doannimble indeed, let me remove 👌

Reply:
Fixed in https://github.com/luongvo/kmm-survey/pull/52/commits/ca8ac3cf7b436aa78b54ec7a90ba6f06bf5e93fd